File: mobile/lib/type/product.dart
class Product {
  final String id;
  final String name;
  final double price;
  final String imageUrl;
  final String reference;
  final String brand;
  final String category;
  final String nutritionalInformation;

  Product({
    required this.id,
    required this.name,
    required this.price,
    required this.imageUrl,
    required this.reference,
    required this.brand,
    required this.category,
    required this.nutritionalInformation,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] ?? '',
      name: (json['name'] != null && json['name'] != '')
          ? json['name']
          : (json['brand'] ?? ''),
      price: (json['price_vat'] ?? 0).toDouble(),
      imageUrl: (json['images'] != null && json['images']['XL'] != null)
          ? json['images']['XL']
          : '',
      reference: json['reference'] ?? '',
      brand: json['brand'] ?? '',
      category: json['category'] ?? '',
      nutritionalInformation: json['nutritional_information'] ?? '',
    );
  }

  @override
  String toString() {
    return 'Product(id: $id, name: $name, price: $price, brand: $brand, category: $category)';
  }
}

File: backend/items/entities/product.go
package entities

type ProductStruct struct {
	Id                     string       `bson:"_id,omitempty" json:"id" validate:"required"`
	Reference              string       `bson:"reference" json:"reference" validate:"required"`
	Images                 ImagesStruct `bson:"images,omitempty" json:"images"`
	PriceVat               float64      `bson:"priceVat" json:"price_vat" validate:"required"`
	PriceNot               float64      `bson:"priceNot" json:"price_not"`
	StockQuantity          float64      `bson:"stockQuantity" json:"stock_quantity"`
	Name                   string       `bson:"name" json:"name" validate:"required"`
	Brand                  string       `bson:"brand" json:"brand"`
	Category               string       `bson:"category" json:"category"`
	NutritionalInformation string       `bson:"nutritionalInformation" json:"nutritional_information"`
	Archived               bool         `bson:"archived" json:"archived"`
}

type ProductOrder struct {
	Id                     string       `bson:"_id,omitempty" json:"id" validate:"required"`
	Reference              string       `bson:"reference" json:"reference" validate:"required"`
	Images                 ImagesStruct `bson:"images,omitempty" json:"images"`
	PriceVat               float64      `bson:"priceVat" json:"price_vat" validate:"required"`
	PriceNot               float64      `bson:"priceNot" json:"price_not"`
	Name                   string       `bson:"name" json:"name" validate:"required"`
	Brand                  string       `bson:"brand" json:"brand"`
	Category               string       `bson:"category" json:"category"`
	NutritionalInformation string       `bson:"nutritionalInformation" json:"nutritional_information"`
}

type ProductBasic struct {
	Reference     string  `json:"reference" validate:"required"`
	PriceVat      float64 `json:"price_vat" validate:"required"`
	PriceNot      float64 `json:"price_not" validate:"required"`
	StockQuantity float64 `json:"stock_quantity" validate:"required"`
}

type StatsProduct struct {
	Name  string `json:"name"`
	Total int32  `json:"total"`
}
